### PR TITLE
Add color encoding to cells

### LINKED ISSUE #49: On-cell color encoding
Color encoding of an attribute
image

### PR DESCRIPTION
**Pull request by gotdairyya**
Closes #49

### CODE REVIEW

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export class View {
   private highlightedNodes: { [key: string]: any[] } = {};
   private columnSelectedNodes: any[] = [];
   private mouseOverEvents: any;
+  private maxVal: number = 0
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:52]: We should probably use a more descriptive variable name for this. You'll also need to propagate this change through the rest of the code.
```suggestion
  private maxNumConnections: number = 0
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 2]
```diff
@@ -168,6 +169,7 @@ export class View {
             }))
           .selectAll('text')
           .style('text-anchor', (d: any, i: number) => i % 2 ? 'end' : 'start');
+
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:172]:
```suggestion
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 3]
```diff
@@ -325,17 +327,28 @@ export class View {
...
+      .attr('height', this.orderingScale.bandwidth())
+      // .style('fill', 'pink') //this will fill the correct squares
+      .style('fill', function(d: { z: number; }) {
+        return cellColorScale(d.z)
+      })
+
+
+
 
     squares
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:348]:
```suggestion
      })

    squares
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 4]
```diff
@@ -325,17 +327,28 @@ export class View {
...
+
     const squares = cells
       .append('rect')
       .attr('x', 0)// d => this.orderingScale(d.x))
       // .filter(d=>{return d.item >0})
       .attr('width', this.orderingScale.bandwidth())
-      .attr('height', this.orderingScale.bandwidth());
-      // .style("fill", 'white')
+      .attr('height', this.orderingScale.bandwidth())
+      // .style('fill', 'pink') //this will fill the correct squares
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:340]:
```suggestion
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 5]
```diff
@@ -325,17 +327,28 @@ export class View {
...
+      })
+
+
+
 
     squares
       .filter((d: any) => d.z === 0)
-      .style('fill-opacity', (d: { z: number; }) => d.z);
+      .style('fill-opacity', (d: { z: number; }) => d.z)
+
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:351]: I don't think you need these changes and they're upsetting the linter. You should roll them back :)

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 6]
```diff
@@ -1231,8 +1244,20 @@ export class View {
...
+        row.forEach(
+          (object: Object) => {
+            if (object.z > this.maxVal) {
+              this.maxVal = object.z
+            }
+          })
+      })
   }
 
+
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1260]:
```suggestion
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 7]
```diff
@@ -1231,8 +1244,20 @@ export class View {
...
+    // find max value of z
+    this.matrix.forEach(
+      (row: Row) => {
+        row.forEach(
+          (object: Object) => {
+            if (object.z > this.maxVal) {
+              this.maxVal = object.z
+            }
+          })
+      })
```
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1257]: The types here aren't quite right. We should discuss how this works on a call

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 8]
```diff
@@ -325,17 +327,28 @@ export class View {
...
-      .attr('height', this.orderingScale.bandwidth());
-      // .style("fill", 'white')
+      .attr('height', this.orderingScale.bandwidth())
+      // .style('fill', 'pink') //this will fill the correct squares
+      .style('fill', function(d: { z: number; }) {
+        return cellColorScale(d.z)
+      })
+
+
+
```
Comment (waxlamp) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:346]:
```suggestion
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts (hunk 5 above)
Comment (waxlamp) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:351]:
```suggestion
      .style('fill-opacity', (d: { z: number; }) => d.z);
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts (hunk 1 above)
Comment (waxlamp) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:52]:
```suggestion
  private maxVal: number = -Infinity;
```

Even though `this.matrix` contains only `0` and `1`, setting the initial `maxVal` this way ensures that the max value computation below can never fail.

File: src/components/AdjMatrix/AdjMatrixMethods.ts (hunk 6 above)
Comment (waxlamp) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1260]:
```suggestion
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts (hunk 7 above)
Comment (JackWilb) [inline src/components/AdjMatrix/AdjMatrixMethods.ts]:
```suggestion
    this.matrix.forEach(
      (row: Array<{z: number, [key: string]: any}>) => {
        row.forEach(
          (cell: {z: number, [key: string]: any}) => {
            if (cell.z > this.maxVal) {
              this.maxVal = cell.z;
            }
          });
      });
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 9]
```diff
@@ -49,6 +49,7 @@ export class View {
   private highlightedNodes: { [key: string]: any[] } = {};
   private columnSelectedNodes: any[] = [];
   private mouseOverEvents: any;
+  private maxVal: number = -Infinity;
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:52]:
```suggestion
  private maxNumConnections: number = -Infinity;
```
More descriptive name per Jack's suggestion

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 10]
```diff
@@ -325,13 +326,20 @@ export class View {
       .attr('height', this.orderingScale.bandwidth())
       .attr('width', this.orderingScale.bandwidth());
 
+    const cellColorScale = d3.scaleLinear<number,number>()
+      .domain([0,this.maxVal])
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:330]:
```suggestion
      .domain([0,this.maxNumConnections])
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 11]
```diff
@@ -1231,6 +1238,17 @@ export class View {
...
       },
     );
+
+    // find max value of z
+    this.matrix.forEach(
+      (row: Row) => {
+        row.forEach(
+          (object: Object) => {
+            if (object.z > this.maxVal) {
+              this.maxVal = object.z
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1248]:
```suggestion
              this.maxNumConnections = object.z
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 12]
```diff
@@ -325,13 +326,19 @@ export class View {
...
       .append('rect')
       .attr('x', 0)// d => this.orderingScale(d.x))
       // .filter(d=>{return d.item >0})
       .attr('width', this.orderingScale.bandwidth())
-      .attr('height', this.orderingScale.bandwidth());
-      // .style("fill", 'white')
+      .attr('height', this.orderingScale.bandwidth())
+      .style('fill', function(d: { z: number; }) {
+        return cellColorScale(d.z)
+      })
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:341]:
```suggestion
      .style('fill', function(d: { z: number; }) => cellColorScale(d.z));
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 13]
```diff
@@ -1231,6 +1236,17 @@ export class View {
...
+    // find max value of z
+    this.matrix.forEach(
+      (row: Row) => {
+        row.forEach(
+          (object: Object) => {
+            if (object.z > this.maxVal) {
+              this.maxNumConnections = object.z
+            }
+          })
+      })
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1249]:
```suggestion
    this.matrix.forEach(
      (row: Array<{ z:number}>) => {
        row.forEach(
          (cell: { z:number}) => {
            if (cell.z > this.maxVal) {
              this.maxNumConnections = cell.z
            }
          });
      });
```

Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1249]:
```suggestion
    this.matrix.forEach(
      (row: Array<{z: number, [key: string]: any}>) => {
        row.forEach(
          (cell: {z: number, [key: string]: any}) => {
            if (cell.z > this.maxVal) {
              this.maxNumConnections = cell.z;
            }
          });
      });
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 14]
```diff
@@ -325,13 +326,17 @@ export class View {
...
+
     const squares = cells
       .append('rect')
       .attr('x', 0)// d => this.orderingScale(d.x))
       // .filter(d=>{return d.item >0})
       .attr('width', this.orderingScale.bandwidth())
-      .attr('height', this.orderingScale.bandwidth());
-      // .style("fill", 'white')
+      .attr('height', this.orderingScale.bandwidth())
+      .style('fill', function(d: { z: number; }) => cellColorScale(d.z));
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:339]:
```suggestion
      .style('fill', (d: { z: number; }) => cellColorScale(d.z));
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 15]
```diff
@@ -1231,6 +1236,17 @@ export class View {
         this.matrix[this.idMap[link.target]][this.idMap[link.source]].z += 1;
       },
     );
+
+    // find max value of z
+    this.matrix.forEach(
+      (row: Array<{z: number, [key: string]: any}>) => {
+        row.forEach(
+          (cell: {z: number, [key: string]: any}) => {
+            if (cell.z > this.maxVal) {
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:1245]:
```suggestion
            if (cell.z > this.maxNumConnections) {
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 16]
```diff
@@ -325,13 +326,17 @@ export class View {
       .attr('height', this.orderingScale.bandwidth())
       .attr('width', this.orderingScale.bandwidth());
 
+    const cellColorScale = d3.scaleLinear<number,number>()
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:329]:
```suggestion
    const cellColorScale = d3.scaleLinear<number,string>()
```

File: src/components/AdjMatrix/AdjMatrixMethods.ts [hunk 17]
```diff
@@ -325,13 +326,17 @@ export class View {
       .attr('height', this.orderingScale.bandwidth())
       .attr('width', this.orderingScale.bandwidth());
 
+    const cellColorScale = d3.scaleLinear<number,string>()
+      .domain([0,this.maxNumConnections])
+      .range(["#feebe2", "#690000"]); // TODO: colors here are arbitrary, change later
```
Comment (gotdairyya) [inline src/components/AdjMatrix/AdjMatrixMethods.ts:331]:
```suggestion
    const cellColorScale = d3.scaleLinear<string>()
      .domain([0, this.maxNumConnections])
      .range(['#feebe2', '#690000']); // TODO: colors here are arbitrary, change later
```

**Pull request merged**